### PR TITLE
Post Video Renderer

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
Added iframe support in Renderhtml by adding @native-html/iframe-plugin .
Added webview in anchor renderer to render markdown-video-link in webview instead of in a separate modal.

### Issue number
Trello: explore iframe native render plugin

### Screenshots

image

### CODE REVIEW

**Comment by noumantahir** [comment]
@feruzm 
Current and new video playback system comparison....

Current system
https://user-images.githubusercontent.com/6298342/150485977-e75a0ce4-efbe-4964-ab17-e50027c22dd8.mov

New System in progress
https://user-images.githubusercontent.com/6298342/150485924-3415490a-7b63-4897-98c5-5ffb716c6dc8.mov

**Comment by noumantahir** [comment]
@aliseyalvi please test and try to perfect the play back of this post, it's a perfect example of a post content containing iframes....

https://ecency.com/hive-172868/@vaketo/-----xgo102be5k8xjugq0drdwpsg4rjvu0gk

**Comment by feruzm** [comment]
> @feruzm Current and new video playback system comparison....
> 
> Current system https://user-images.githubusercontent.com/6298342/150485977-e75a0ce4-efbe-4964-ab17-e50027c22dd8.mov
> 
> New System in progress https://user-images.githubusercontent.com/6298342/150485924-3415490a-7b63-4897-98c5-5ffb716c6dc8.mov

I like new system 👏

**Comment by aliseyalvi** [comment]
@noumantahir 

- Added iframe support. Replaced iframe with custom iframe renderer using webview.  https://github.com/ecency/ecency-mobile/pull/2154/commits/68a94dd2c13541ab5c11b77430c98f66fc35f9d5
- Wrapped source with video tag if  source is video file which fixes the autoplay issue.
- changed webview source for supporting all type of videos like Youtube, vimeo etc. https://github.com/ecency/ecency-mobile/pull/2154/commits/8b4e2ea013c661cb4f03d881e255eb3fdadfd670
- Some Youtube videos are showing not available error. Might be due to privacy issue or could be other bug. 
image

**Comment by noumantahir** [comment]
@aliseyalvi after assessing your solution, it seems we should be able to easily repurpose `videoPlayerSheet.tsx` to achieve our goal instead of rewriting webviews.

We can rename VideoPlayerSheet to simple VideoPlayer and call it with videoUrl for regular videos and with youtube id in case of youtube and that should also fix the youtube issue you are facing.

Another suggestion from my side would be to skip comments from using new video playback system as they can overload the system.

File: src/components/postHtmlRenderer/postHtmlRenderer.tsx
Comment (noumantahir) [inline src/components/postHtmlRenderer/postHtmlRenderer.tsx:165]: for youtube... the class is `markdown-video-link-youtube` and attributes are as follows

data-youtube: contains video id
data-start-time: contains start time of video (optional)

pass this data to youtube iframe implemented inside videoPlayerSheet.tsx and it should play without issues.

**Review by noumantahir — COMMENTED** [review]
It's working alright, video source with extension is an exception we can accept for now.... Do checkout my comment regarding how we can repurpose videoPlayerSheet to avoid code duplications.

**Comment by aliseyalvi** [comment]
@noumantahir created videoPlayer component which handles youtube and other videos as well. Iframe renderer has kept as it is because it handles source differently. review the changes https://github.com/ecency/ecency-mobile/pull/2154/commits/9c3f12e06e4b52dac0c3328382f12dbfc6f828bc

**Comment by noumantahir** [comment]
**Updates**
Unified Video Player Component
Using Sheet Style player for comments and inline videos
Disabled autoplay youtube for post videos
Simplified anchorRenderer

@aliseyalvi please go through changes and test if I missed anything that was intentional
@feruzm you may review as well

**Review by noumantahir — APPROVED** [review]
All looks good to me now... require more testing and verifying.

**Comment by noumantahir** [comment]
@feruzm this PR is awaiting your approval before merging

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**
